### ISSUE TITLE
check_arrays could support memmaps

### ISSUE THREAD

**Issue by hugorichard** [mvlearn/mvlearn#289]
When multiple datasets are used, memory can be overloaded easily unless one is careful about the implementation.
I think it makes sense to assume that:
- Reduced data can hold in memory
- One dataset can hold in memory
I would like to modify the check_arrays function so that it supports memory maps. I may also change some code in GroupPCA so that it uses reduced amount of memory.
Do you think it is a good idea ?

**Comment by rflperry** [comment]
Yes I think this could be useful. I've run into such an issue before applying groupPCA like algorithms to many correlation matrices at once. I'm not very familiar with memory maps. It may be that a `partial_fit` function could solve this as it's a two-step algorithm and holding the initial decompositions is probably not a problem.
